openreview/venue_request/process/matching_pre_process.py

```diff
@@ -1,4 +1,21 @@
 def process(client, note, invitation):
 
     if 'No' not in note.content['compute_affinity_scores'] and 'upload_affinity_scores' in note.content:
-        raise openreview.OpenReviewException('Either upload your own affinity scores or select affinity scores computed by OpenReview')
\ No newline at end of file
+        raise openreview.OpenReviewException('Either upload your own affinity scores or select affinity scores computed by OpenReview')
+
+    matching_group = note.content['matching_group']
+    matching_notes = client.get_all_notes(invitation=invitation.id, sort='tmdate:desc')
+
+    if matching_notes:
+        latest_matching_note = None
+        for i, matching_note in enumerate(matching_notes):
```

Comment:
I don't think you need to use the index here:

```
for matching_note in matching_notes:
       if ....:
           latest_matching_note  =matching_note
```

A nicer way to do this and to avoid the `break` sentence is to use a lambda function:

```
matching_group_notes = [matching_note for matching_note in matching_notes if matching_note.content['matching_group'] == 'matching_group']

if matching_group_notes:
    status_note = client.get_all_notes(invitation= invitation.id.replace('Paper_Matching_Setup', 'Paper_Matching_Setup_Status'), replyto=matching_group_notes[0].id)
    if not status_note:
       ....
```

    

```

Reply:
You're right, that is much better. Updated now.